api/src/pcapi/utils/secrets.py

```diff
@@ -44,10 +44,31 @@ def dump_secret_keys() -> str:
     return yaml.dump(yaml_content)
 
 
+def _check_secret_list_has_been_initialized() -> None:
+    if not SECRET_KEYS:
+        raise ValueError("The list of secrets has not been initialized.")
+
+
 @blueprint.cli.command("print_secret_keys")
 def print_secret_keys() -> None:
     """
     Prints secret's keys list in yaml format.
     This output is used in deployment steps.
     """
-    print(dump_secret_keys(), file=sys.stdout)
+    _check_secret_list_has_been_initialized()
+
+    print(dump_secret_keys())
+
+
+@blueprint.cli.command("check_secrets")
+def check_secrets() -> None:
+    """Make sure that all secrets are defined and non-empty."""
+    _check_secret_list_has_been_initialized()
+
+    missing = set()
+    for secret in SECRET_KEYS:
+        if not os.environ.get(secret):
```

Comment:
Que se passe-t-il si la valeur du secret est `""` ou `0` ou `False` ou `[]` ?

Reply:
Les variables d'environnement sont des chaînes de caractères. Si une variable d'environnement est vraiment vide, c'est qu'elle n'est pas définie, ce qui n'est pas normal.

Reply:
Ah oui, évidemment des `str` 🤦 .
Les secrets qui définissent `FINANCE_OVERRIDE_PRICING_ORDERING_ON_PRICING_POINTS` sont souvent une chaine vide il me semble, donc `not ""` sera évalué à `True` ?

Reply:
> Les secrets qui définissent FINANCE_OVERRIDE_PRICING_ORDERING_ON_PRICING_POINTS sont souvent une chaine vide il me semble, donc not "" sera évalué à True ?

Non, c'est toujours au moins un espace, car les secrets sur GCP ne peuvent pas être une chaîne vide. L'espace seul est ensuite transformé en chaîne vide par le `strip()` ici : https://github.com/pass-culture/pass-culture-main/blob/2d4681fc427fe7e5b71ef806e921e1152900b9c6/api/src/pcapi/utils/secrets.py#L27